common/windows/delphi/general/Upload_Settings.pas

```diff
@@ -179,4 +181,9 @@ function URLPath_PackageDownload(const PackageID, BCP47: string; IsUpdate: Boole
   Result := Format(URLPath_PackageDownload_Format, [URLEncode(PackageID), URLEncode(CKeymanVersionInfo.Tier), URLEncode(BCP47), IsUpdateInt]);
 end;
 
+function URL_KmcMessage(id: string): string;
+begin
+   Result := S_HelpKeymanCom + '/developer/'+GetMajorMinorVersionString+'/reference/messages/'+id.ToLower;
```

Comment:
I now realize that this violates our earlier rule of always using /go/ urls in code. Thoughts?

Reply:
Note that the compiler error message details even in this PR also contain long-form URLs, so perhaps we need to revisit this rule?

Reply:
Looking at what we already have with `kmc message`, I think we should go with `https://kmn.sh/km#####` as our redirect (already baked in). It's also a redirect with intent for URLs to be permanent so if help site layout changes then it will just mean updating the redirect.

I am not so sure about URLs embedded in message details. Am considering leaving those as-is, and declaring them to be an exception to the rule, because otherwise we have to add a lot of redirects with little benefit.

Reply:
I've made validation of hard-coded URLs in kmc messages into a future issue #13164; FYI @darcywong00